### ISSUE TITLE
Don't duplicate samples when selecting resulting cases for event participants and contacts

### ISSUE THREAD

**Issue by adinaflorea9** [hzi-braunschweig/SORMAS-Project#9632]
### Bug Description

The duplicate detection for existing cases is triggered twice when creating new cases for event participants.

If in the first 'Pick or create case' pop-up the user selects 'Pick an existing case' the sample is duplicated in the case page, when creating the case from an event participant or a contact.

### Steps to Reproduce

1. Open an existing event that has no disease set and create a new event participant;
2. Add a sample with positive pathogen test result for Covid-19;
-> pop-up to create a new Covid-19 case is triggered;
3. Fill in the mandatory information and save the new case;
4. Navigate back to the event from step 1 and set the disease of the event to Covid-19;
5. Navigate to the 'Event Participants' list page and for the event participant from step 1 select the existing case

### Actual Behavior

a. ~~'Pick or create case' pop-up is triggered a second time.~~

b. Alternatively, if at step 6 the user selects 'Pick an existing case', the actual behavior is that there is only one case for this event participant, but the existing sample is duplicated (even though no other sample or pathogen test result were created by the user when trying to create the second Covid-19 case).

Behavior from this section is also reproducible for contacts (if a contact is first created with the disease Covid-19; a sample with positive pathogen test result for Monkeypox is added to it, the new Monkeypox case is created; then the disease of the contact is changed to Monkeypox and the user creates a resulting case for the contact manually from the 'Create a case for this contact person' button and selects 'Pick an existing case' when the duplicate detection for cases is triggered).

### Expected Behavior

~~The duplicate detection for cases is triggered only once (only the first 'Pick or create case' pop-up should be triggered).~~

When the user selects 'Pick an existing case' the sample from the case page should not be duplicated.

### Screenshots

Steps to reproduce - Event Participant - 'Pick or create case' pop-up triggered twice:
![EventParticipant](https://user-images.githubusercontent.com/96909231/175930475-f13fbdd7-5d2d-4a6f-8434-a6a47f380bc8.gif)

Steps to reproduce - Event Participant - sample is duplicated:
![DuplicatedSample](https://user-images.githubusercontent.com/96909231/175931632-29e4d502-60a9-44a1-b88c-8ca4602e62af.gif)

Steps to reproduce - Contact - sample is duplicated:
![ContactSample](https://user-images.githubusercontent.com/96909231/175933076-3be695be-8296-4ab0-bfb8-e6d68296d2a1.gif)

### System Details

* Device: Windows 10
* SORMAS version: 1.74.0-SNAPSHOT(e3d0ece) 
* Android version/Browser: Chrome
* Server URL: local machine
* User Role: Admin+NatUser

### Additional Information

Behavior is similar to the one reported in https://github.com/hzi-braunschweig/SORMAS-Project/issues/8395 where adding 2 pathogen test results led to the duplication of the sample. The current separate issue was logged as the steps to reproduce are different than those from #8395 and the duplicate detection for cases does not lead currently to any error message.

**Comment by MateStrysewske** [comment]
@adinaflorea9 @JaquM-HZI We have to decide what the expected behaviour should be here, because in my opinion, the duplicate detection popup appearing twice is absolutely correct (it SHOULD trigger when clicking on "Create case" because it detects that there's a case in the database that matches this person, and it SHOULD again trigger after filling out the case information because, again, it detects that there's already such a case in the database). The only thing that might be worth looking into is how samples are handled here, because indeed it doesn't seem to make sense that they're duplicated. What's your take on this?

**Comment by adinaflorea9** [comment]
@MateStrysewske Thank you for explaining the logic behind the triggering of the duplicate detection in this instance. Based on your explanation, I agree that it is alright that the pop-up is triggered twice, and that the handling of samples should remain in the focus of the ticket.

**Comment by JaquM-HZI** [comment]
@MateStrysewske would agree as well. Please focus on samples.

**Comment by roxanamlendea** [comment]
I started looking over the ticket and I'm reopening it because of the following finding:

1. Create an Event with Covid 19 selected as Disease;
2. Create an Event Participant;
3. From the Event Participants page, create a Case for the Event Participant from step 2;
4. Navigate back to the Event and set the Event's Disease to null;
5. Open the Event Participant's page and add a Sample with a positive Pathogen Test with Covid 19 as disease;
6. In the following pop-up select 'Yes'
![image](https://user-images.githubusercontent.com/89522688/176853087-a9b09298-9439-4aab-a809-7bc2ff42fb0c.png)
7. Fill in the mandatory fields;
8. In the following pop-up, select 'Pick an existing Case';
![image](https://user-images.githubusercontent.com/89522688/176853208-d76547b3-2a06-4d5a-980d-5702146b308f.png)

-> The Sample is not associated to the Case, only to the Event Participant.

**Comment by MateStrysewske** [comment]
@roxanamlendea Just checked the code and this behaviour isn't related to this bug, it's basically something that we haven't considered yet (because it's also very unlikely to happen); may I ask you to open a separate change request and document your observations there? Title would be "Take over samples when creating unrelated cases from event participants".

**Comment by roxanamlendea** [comment]
> @roxanamlendea Just checked the code and this behaviour isn't related to this bug, it's basically something that we haven't considered yet (because it's also very unlikely to happen); may I ask you to open a separate change request and document your observations there? Title would be "Take over samples when creating unrelated cases from event participants".

Sure, I'll open a change request for this then.

**Comment by roxanamlendea** [comment]
Verified ticket on local environment using the latest version of Sormas 1.74.0-SNAPSHOT (77d24ce) from the development branch.
